src/nrniv/nrncore_write/data/cell_group.cpp

```diff
@@ -311,6 +311,9 @@ void CellGroup::datumindex_fill(int ith, CellGroup& cg, DatumIndices& di, Memb_l
             } else if (dmap[j] == -10) {  // fornetcon
                 etype = -10;
                 eindex = 0;
+            } else if (dmap[j] == -11) {  // random
```

Comment:
another good candidate for a switch case

Reply:
Not really:
```
            } else if (dmap[j] > 0 && dmap[j] < 1000) {  // double* into eion type data
...
            } else if (dmap[j] > 1000) {  // int* into ion dparam[xxx][0]
```
are what you'll find in later parts of the tree.

Personally, I don't like `switch` statements, because they only work for integers and you can only check for equality, unless you abuse it's most twisted feature that is "falls" through, i.e. forget an `break` and things go wrong. Finally, the scoping rules don't allow local variables, e.g.
```
switch(c) {
  case 1:
    double x = 42;
  case 2:
    double x = 43;
}
```
fail to compile and need an extra pair of `{}`.

Reply:
There will another chance to devise a decent implementation that elegantly and uniformly handles semantics when the ```NEURON { VECTOR vec1, vec2...}``` statement is added. Or it could be done as a separate pull request.